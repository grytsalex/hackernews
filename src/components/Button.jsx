import React from "react";
import PropTypes from "prop-types";

export const Button = ({ onClick, className, children, customStyles }) => (
  <button onClick={onClick} className={className} type="button" style={{...customStyles}}>
    {children}
  </button>
);

Button.defaultProps = {
  className: "",
};

Button.propTypes = {
  onClick: PropTypes.func.isRequired,
  className: PropTypes.string,
  children: PropTypes.node.isRequired,
};
