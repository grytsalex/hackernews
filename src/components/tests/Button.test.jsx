import React from "react";
import renderer from "react-test-renderer";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

import { Button } from "../Button";

Enzyme.configure({ adapter: new Adapter() });

describe("Button", () => {
  const props = {
    onClick: jest.fn(),
    children: "Press",
    className: "",
  };

  test("should have snapshot", () => {
    const component = renderer.create(<Button {...props} />);

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should call onClick when press on button", () => {
    const element = shallow(<Button {...props} />);
    const button = element.find("button");
    button.simulate("click");
    expect(props.onClick).toHaveBeenCalledTimes(1);
  });
});
