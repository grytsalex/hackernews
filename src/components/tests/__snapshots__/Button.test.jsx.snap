// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should have snapshot 1`] = `
<button
  className=""
  type="button"
>
  Press
</button>
`;
